File: android/app/src/main/java/com/smartexpense/android/domain/usecase/auth/LoginUseCase.java
package com.smartexpense.android.domain.usecase.auth;

import com.smartexpense.android.data.remote.dto.request.LoginRequestDto;
import com.smartexpense.android.data.remote.dto.response.AuthResponseDto;
import com.smartexpense.android.domain.repository.AuthRepository;
import com.smartexpense.android.domain.repository.ResultCallback;

public class LoginUseCase {
    private final AuthRepository repository;

    public LoginUseCase(AuthRepository repository) {
        this.repository = repository;
    }

    public void execute(String email, String password, ResultCallback<AuthResponseDto> callback) {
        LoginRequestDto request = new LoginRequestDto(email, password);
        repository.login(request, callback);
    }
}

File: android/app/src/main/kotlin/com/smartexpense/android/domain/usecase/auth/LoginUseCase.kt
package com.smartexpense.android.domain.usecase.auth

import com.smartexpense.android.data.remote.dto.request.LoginRequestDto
import com.smartexpense.android.data.remote.dto.response.AuthResponseDto
import com.smartexpense.android.domain.repository.AuthRepository

class LoginUseCase(private val repository: AuthRepository) {
    suspend fun execute(email: String, password: String): Result<AuthResponseDto> {
        val request = LoginRequestDto(email, password)
        return repository.login(request)
    }
}
